src/directives/badge/badge.directive.ts

```diff
@@ -0,0 +1,204 @@
+
+    /**
+     * Set if the badge overlaps parent content or flows after parent
+     */
+    @HostBinding('class.ux-badge-overlap')
+    @Input()
+    badgeOverlap: boolean = false;
+
+    /**
+     * Max value that can be displayed - if string measures no. of characters,
+     * if number checks against exact number not no. of characters
+     */
+    @Input() badgeMaxValue: number;
+
+    /**
+     * Badge size (based on CSS styles)
+     */
+    private _defaultBadgeSize: BadgeSize = 'medium';
+
+    @Input()
+    private badgeSize: BadgeSize = this._defaultBadgeSize;
+
+    /**
+     * Hide badge from view
+     */
+    @HostBinding('class.ux-badge-hidden')
+    @Input()
+    badgeHidden: boolean = false;
+
+    constructor(
+        private readonly _element: ElementRef<HTMLElement>,
+        private readonly _renderer: Renderer2,
+        private readonly _colorService: ColorService,
+        private readonly _contrastService: ContrastService
+    ) {}
+
+    ngOnChanges(changes: SimpleChanges): void {
+        // get display friendly version of text based on max length and type of val;
+        if (changes.badgeContent) {
+            let finalText: string = changes.badgeContent.currentValue || null;
+
+            if (finalText && this.badgeMaxValue) {
+                if (this._isNumber && changes.badgeContent.currentValue > this.badgeMaxValue) {
+                    finalText = `${this.badgeMaxValue}+`;
+                } else if (changes.badgeContent.currentValue.length > this.badgeMaxValue) {
+                    this._truncatedText = true;
+                    finalText = `${changes.badgeContent.currentValue.substr(0, this.badgeMaxValue)}&hellip;`;
+                } else {
+                    this._truncatedText = false;
+                }
+            }
+
+            this._badgeDisplayContent = finalText;
+        }
+
+        // if text is only set, set aria label from it
+        if (changes.badgeContent && !this.badgeAriaDescription) {
+            this.badgeAriaDescription = changes.badgeContent.currentValue;
+        }
+
+        this.updateBadge();
+    }
+
+    ngOnDestroy(): void {
+        if (this._renderer.destroyNode) {
+            this._renderer.destroyNode(this._badgeElement);
+        }
+    }
+
+    private updateBadge(): HTMLElement {
+        const badgeElement: HTMLSpanElement = this._renderer.createElement('span');
+        this.clearExisting();
+
+        badgeElement.classList.add(this._className);
+        badgeElement.style.setProperty('display', 'none');
+
+        badgeElement.innerHTML = this._badgeDisplayContent;
+        badgeElement.style.setProperty('color', this.determineContentTextColor().toHex());
+
+        if (this.badgeAriaDescription) {
+            badgeElement.setAttribute('aria-describedby', this.badgeAriaDescription);
+        }
+
+        if (this._truncatedText) {
+            badgeElement.setAttribute('title', this._badgeContent);
```

Comment:
Hmm, so this is adding a browser tooltip if there is truncation occurring. We have avoided the browser tooltips in all our other components/directives as there are MF styled tooltips, however I believe they are currently only a directive so we can't programmatically instantiate one here.

@donna-marijne What are your thoughts here?

Reply:
I did a small amount of digging on this since I know that a lot of sites do use `title` as a cheap tooltip.

https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title#Accessibility_concerns

These points lead me to believe that we should not add something like this unless it becomes a specific requirement. We should bear in mind that the truncation is fully under the implementor's control in this case, so it could be argued that it's up to the application itself to make provisions if the text length routinely exceeds the configured maximum (e.g. by adding a tooltip to the host element).

Reply:
This is now removed